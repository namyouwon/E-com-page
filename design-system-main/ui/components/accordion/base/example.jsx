// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import uniqueId from 'lodash.uniqueid';
import Accordion, { AccordionSection } from '../';
import { ActionOverflow } from '../../../components/menus/action-overflow/example';
import Card from '../../cards/';

export const AccordionExample = () => (
  <Accordion>
    <AccordionSection
      summary="Accordion summary"
      isOpen
      referenceId={uniqueId('referenceId-')}
    >
      Accordion details - A
    </AccordionSection>
    <AccordionSection
      summary="Accordion summary"
      referenceId={uniqueId('referenceId-')}
    >
      Accordion details - B
    </AccordionSection>
    <AccordionSection
      summary="Accordion summary"
      referenceId={uniqueId('referenceId-')}
    >
      Accordion details - C
    </AccordionSection>
  </Accordion>
);

export default <AccordionExample />;

export let states = [
  {
    id: 'section-one-open',
    label: 'Section one open',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'section-two-open',
    label: 'Section two open',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'section-three-open',
    label: 'Section three open',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'multiple-sections-open',
    label: 'Multiple sections open',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  }
];

export let examples = [
  {
    id: 'long-heading-truncated',
    label: 'Long Section Heading with Truncation for larger screens',
    demoStyles: 'max-width: 500px;',
    storybookStyles: true,
    element: (
      <Card>
        <Accordion>
          <AccordionSection
            summary="Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary"
            isOpen
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - A
          </AccordionSection>
          <AccordionSection
            summary="Accordion summary"
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - B
          </AccordionSection>
          <AccordionSection
            summary="Accordion summary"
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - C
          </AccordionSection>
        </Accordion>
      </Card>
    )
  },
  {
    id: 'nested-accordions',
    label: 'Nested Accordions',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          <Accordion>
            <AccordionSection
              summary="Nested accordion summary"
              isOpen
              referenceId={uniqueId('referenceId-')}
            >
              Accordion details - A
            </AccordionSection>
            <AccordionSection
              summary="Nested accordion summary"
              referenceId={uniqueId('referenceId-')}
            >
              Accordion details - B
            </AccordionSection>
          </Accordion>
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'styled',
    label: 'Wrapped in Card',
    element: (
      <div className="slds-card">
        <Accordion>
          <AccordionSection
            summary="Accordion summary"
            isOpen
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - A
          </AccordionSection>
          <AccordionSection
            summary="Accordion summary"
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - B
          </AccordionSection>
          <AccordionSection
            summary="Accordion summary"
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - C
          </AccordionSection>
        </Accordion>
      </div>
    )
  },
  {
    id: 'has-action-overflow',
    label: 'Has Action Overflow Menu',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          actionMenu={
            <ActionOverflow position="right" dropdownIsOpen={false} />
          }
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          actionMenu={
            <ActionOverflow position="right" dropdownIsOpen={false} />
          }
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          actionMenu={
            <ActionOverflow position="right" dropdownIsOpen={false} />
          }
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'section-background-color',
    label: 'Accordion Section - Background Color',
    demoStyles: '--slds-c-accordion-summary-color-background: lightblue',
    storybookStyles: true,
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'section-background-color',
    label: 'Accordion Section - Text Color',
    demoStyles:
      '--slds-c-accordion-heading-text-color: green; --slds-c-accordion-heading-text-color-hover: red',
    storybookStyles: true,
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  }
];
