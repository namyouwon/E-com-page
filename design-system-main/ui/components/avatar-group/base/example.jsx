// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import {
  AvatarGroup,
  AvatarGroupContent,
  AvatarGrouped,
  AvatarImage,
  AvatarGroupedIcon,
  AvatarGroupedUserInitials
} from '../';

export const ContextInversed = props => (
  <div
    className="demo-only"
    style={{ padding: '0.5rem', background: '#16325c' }}
  >
    {props.children}
  </div>
);

export default [
  {
    id: 'default',
    label: 'Default',
    element: (
      <AvatarGroup>
        <AvatarGroupContent type="image" />
      </AvatarGroup>
    )
  }
];

export const examples = [
  {
    id: 'with-icons',
    label: 'With icons',
    element: (
      <AvatarGroup>
        <AvatarGroupContent type="icon" />
      </AvatarGroup>
    )
  },
  {
    id: 'with-user-initials',
    label: 'With initials',
    element: (
      <AvatarGroup>
        <AvatarGroupContent type="user-initials" />
      </AvatarGroup>
    )
  },
  {
    id: 'with-user-initials-inversed',
    label: 'With initials inversed',
    demoStyles: 'padding: 0.5rem; background: #16325c',
    storybookStyles: true,
    element: (
      <AvatarGroup>
        <AvatarGroupContent type="user-initials" isInverse />
      </AvatarGroup>
    )
  },
  {
    id: 'with-image-and-icon',
    label: 'With image and icon',
    element: (
      <AvatarGroup>
        <AvatarGrouped isPrimary>
          <AvatarImage />
        </AvatarGrouped>
        <AvatarGrouped>
          <AvatarGroupedIcon />
        </AvatarGrouped>
      </AvatarGroup>
    )
  },
  {
    id: 'with-icon-and-user-initials',
    label: 'With icon and user initials',
    element: (
      <AvatarGroup>
        <AvatarGrouped isPrimary>
          <AvatarGroupedIcon />
        </AvatarGrouped>
        <AvatarGrouped>
          <AvatarGroupedUserInitials />
        </AvatarGrouped>
      </AvatarGroup>
    )
  },
  {
    id: 'with-user-initials-and-image',
    label: 'With user initials and image',
    element: (
      <AvatarGroup>
        <AvatarGrouped isPrimary>
          <AvatarGroupedUserInitials />
        </AvatarGrouped>
        <AvatarGrouped>
          <AvatarImage />
        </AvatarGrouped>
      </AvatarGroup>
    )
  },
  {
    id: 'x-small-image',
    label: 'X-Small',
    element: (
      <AvatarGroup size="x-small">
        <AvatarGroupContent type="image" />
      </AvatarGroup>
    )
  },
  {
    id: 'small-image',
    label: 'Small',
    element: (
      <AvatarGroup size="small">
        <AvatarGroupContent type="image" />
      </AvatarGroup>
    )
  },
  {
    id: 'medium-image',
    label: 'Medium',
    element: (
      <AvatarGroup size="medium">
        <AvatarGroupContent type="image" />
      </AvatarGroup>
    )
  },
  {
    id: 'large-image',
    label: 'Large',
    element: (
      <AvatarGroup size="large">
        <AvatarGroupContent type="image" />
      </AvatarGroup>
    )
  },
  {
    id: 'x-small-icon',
    label: 'X-Small With icon',
    element: (
      <AvatarGroup size="x-small">
        <AvatarGroupContent type="icon" />
      </AvatarGroup>
    )
  },
  {
    id: 'small-icon',
    label: 'Small With icon',
    element: (
      <AvatarGroup size="small">
        <AvatarGroupContent type="icon" />
      </AvatarGroup>
    )
  },
  {
    id: 'medium-icon',
    label: 'Medium With icon',
    element: (
      <AvatarGroup size="medium">
        <AvatarGroupContent type="icon" />
      </AvatarGroup>
    )
  },
  {
    id: 'large-icon',
    label: 'Large With icon',
    element: (
      <AvatarGroup size="large">
        <AvatarGroupContent type="icon" />
      </AvatarGroup>
    )
  },
  {
    id: 'x-small-user-initials',
    label: 'X-Small With initials',
    element: (
      <AvatarGroup size="x-small">
        <AvatarGroupContent type="user-initials" />
      </AvatarGroup>
    )
  },
  {
    id: 'small-user-initials',
    label: 'Small With initials',
    element: (
      <AvatarGroup size="small">
        <AvatarGroupContent type="user-initials" />
      </AvatarGroup>
    )
  },
  {
    id: 'medium-user-initials',
    label: 'Medium With initials',
    element: (
      <AvatarGroup size="medium">
        <AvatarGroupContent type="user-initials" />
      </AvatarGroup>
    )
  },
  {
    id: 'large-user-initials',
    label: 'Large With initials',
    element: (
      <AvatarGroup size="large">
        <AvatarGroupContent type="user-initials" />
      </AvatarGroup>
    )
  }
];
