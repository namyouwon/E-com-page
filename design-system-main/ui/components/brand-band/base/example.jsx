// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';

import BrandBand from '../';

export default <BrandBand size="medium" />;

export let examples = [
  {
    id: 'brand-band-small',
    label: 'Small',
    element: <BrandBand size="small" />
  },
  {
    id: 'brand-band-large',
    label: 'Large',
    element: <BrandBand size="large" />
  },
  {
    id: 'brand-band-no-image',
    label: 'No Image',
    element: <BrandBand size="none" />
  },
  {
    id: 'brand-band-large-group',
    label: 'Large Group',
    element: <BrandBand size="large" image="group" />
  },
  {
    id: 'brand-band-large-user',
    label: 'Large User',
    element: <BrandBand size="large" image="user" />
  }
];
