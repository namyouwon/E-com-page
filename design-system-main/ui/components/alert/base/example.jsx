// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import Alert from '../';
import { UtilityIcon } from '../../icons/base/example';
import EmptyLink from '../../../shared/empty-link/';
import Heading from '../../../shared/heading/';

export default [
  {
    id: 'default',
    label: 'Default',
    element: (
      <Alert>
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="user"
        />
        <Heading level="2">
          Logged in as John Smith (johnsmith@acme.com).{' '}
          <EmptyLink>Log out</EmptyLink>
        </Heading>
      </Alert>
    )
  }
];

export let states = [
  {
    id: 'theme-info',
    label: 'Default (Theme Class  - Deprecated)',
    element: (
      <Alert isLegacy>
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="user"
        />
        <Heading level="2">
          Logged in as John Smith (johnsmith@acme.com).{' '}
          <EmptyLink>Log out</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'warning',
    label: 'Warning',
    element: (
      <Alert type="warning">
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="warning"
        />
        <Heading level="2">
          Your browser is outdated. Your Salesforce experience may be degraded.{' '}
          <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'theme-warning',
    label: 'Warning (Theme Class  - Deprecated)',
    element: (
      <Alert type="warning" isLegacy>
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="warning"
        />
        <Heading level="2">
          Your browser is outdated. Your Salesforce experience may be degraded.{' '}
          <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'error',
    label: 'Error',
    element: (
      <Alert type="error">
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="error"
        />
        <Heading level="2">
          Your browser is currently not supported. Your Salesforce may be
          degraded. <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'theme-error',
    label: 'Error (Theme Class  - Deprecated)',
    element: (
      <Alert type="error" isLegacy>
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="error"
        />
        <Heading level="2">
          Your browser is currently not supported. Your Salesforce may be
          degraded. <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'offline',
    label: 'Offline',
    element: (
      <Alert type="offline">
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="offline"
        />
        <Heading level="2">
          You are in offline mode. <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'theme-offline',
    label: 'Offline (Theme Class  - Deprecated)',
    element: (
      <Alert type="offline" isLegacy>
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="offline"
        />
        <Heading level="2">
          You are in offline mode. <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  }
];
