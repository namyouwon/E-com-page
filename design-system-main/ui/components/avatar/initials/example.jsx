// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import { Avatar } from '../base/example';
import { AvatarInitials } from '../';

export default [
  {
    id: 'default',
    label: 'Default',
    element: (
      <Avatar>
        <AvatarInitials
          iconClassName="slds-icon-standard-account"
          title="company name"
          initials="Ac"
        />
      </Avatar>
    )
  }
];

export let examples = [
  {
    id: 'linked-avatar-initials',
    label: 'Wrapped in a link',
    element: (
      <a href="#" onClick={e => e.preventDefault()}>
        <Avatar>
          <AvatarInitials
            iconClassName="slds-icon-standard-account"
            title="company name"
            initials="Ac"
          />
        </Avatar>
      </a>
    )
  }
];
