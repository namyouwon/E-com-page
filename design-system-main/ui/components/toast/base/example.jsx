// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import Toast from '../';
import { UtilityIcon } from '../../icons/base/example';

export default [
  {
    id: 'default',
    label: 'Base - default',
    element: (
      <Toast type="info" containerClassName="slds-is-relative">
        <UtilityIcon
          containerClassName="slds-m-right_small slds-no-flex slds-align-top"
          className="slds-icon_small"
          assistiveText={false}
          symbol="info"
        />
        <div className="slds-notify__content">
          <h2 className="slds-text-heading_small">
            26 potential duplicate leads were found.{' '}
            <a href="#" onClick={e => e.preventDefault()}>
              Select Leads to Merge
            </a>
          </h2>
        </div>
      </Toast>
    )
  }
];

export let states = [
  {
    id: 'success',
    label: 'Success (Theme Class)',
    element: (
      <Toast type="success" containerClassName="slds-is-relative">
        <UtilityIcon
          containerClassName="slds-m-right_small slds-no-flex slds-align-top"
          className="slds-icon_small"
          assistiveText={false}
          symbol="success"
        />
        <div className="slds-notify__content">
          <h2 className="slds-text-heading_small ">
            Account{' '}
            <a href="#" onClick={e => e.preventDefault()}>
              ACME - 100
            </a>{' '}
            widgets was created.
          </h2>
        </div>
      </Toast>
    )
  },
  {
    id: 'warning',
    label: 'Warning (Theme Class)',
    element: (
      <Toast type="warning" containerClassName="slds-is-relative">
        <UtilityIcon
          containerClassName="slds-m-right_small slds-no-flex slds-align-top"
          className="slds-icon_small"
          assistiveText={false}
          symbol="warning"
        />
        <div className="slds-notify__content">
          <h2 className="slds-text-heading_small ">
            Can’t share file “report-q3.pdf” with the{' '}
            <a href="#" onClick={e => e.preventDefault()}>
              selected users
            </a>
            .
          </h2>
        </div>
      </Toast>
    )
  },
  {
    id: 'error',
    label: 'Error (Theme Class)',
    element: (
      <Toast type="error" containerClassName="slds-is-relative">
        <UtilityIcon
          containerClassName="slds-m-right_small slds-no-flex slds-align-top"
          className="slds-icon_small"
          assistiveText={false}
          symbol="error"
        />
        <div className="slds-notify__content">
          <h2 className="slds-text-heading_small ">
            Can’t save lead “Sally Wong” because{' '}
            <a href="#" onClick={e => e.preventDefault()}>
              another lead
            </a>{' '}
            has the same name.
          </h2>
        </div>
      </Toast>
    )
  },
  {
    id: 'error-with-details',
    label: 'Error With Details (Theme Class)',
    element: (
      <Toast type="error" containerClassName="slds-is-relative">
        <UtilityIcon
          containerClassName="slds-m-right_small slds-no-flex slds-align-top"
          className="slds-icon_small"
          assistiveText={false}
          symbol="error"
        />
        <div className="slds-notify__content">
          <h2 className="slds-text-heading_small">
            You've encountered some errors when trying to save edits to Samuel
            Smith.
          </h2>
          <p>
            Here's some detail of{' '}
            <a href="#" onClick={e => e.preventDefault()}>
              what happened
            </a>
            , being very descriptive and transparent.
          </p>
        </div>
      </Toast>
    )
  }
];

export let examples = [
  {
    id: 'small',
    label: 'Small Column',
    demoStyles: 'width: 25rem',
    storybookStyles: true,
    element: (
      <div className="slds-region_narrow">
        <Toast type="info" containerClassName="slds-is-relative">
          <div className="slds-notify__content">
            <h2 className="slds-text-heading_small">
              26 potential{' '}
              <a href="#" onClick={e => e.preventDefault()}>
                duplicate leads
              </a>{' '}
              were found.
            </h2>
          </div>
        </Toast>
      </div>
    )
  }
];
