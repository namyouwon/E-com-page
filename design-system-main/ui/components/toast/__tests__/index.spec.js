/* eslint-env jest */
import React from 'react';
import createHelpers from '../../../../jest.helpers';
import Toast from '../index';
import { UtilityIcon } from '../../icons/base/example';

const { matchesMarkup } = createHelpers(__dirname);

describe('Toast', () => {
  it('renders a default toast for info', () =>
    matchesMarkup(
      <Toast>
        <div className="slds-notify__content">
          <h2 className="slds-text-heading_small">
            26 potential duplicate leads were found.{' '}
            <a href="#" onClick={e => e.preventDefault()}>
              Select Leads to Merge
            </a>
          </h2>
        </div>
      </Toast>
    ));

  it('renders an info toast', () =>
    matchesMarkup(
      <Toast type="info">
        <div className="slds-notify__content">
          <h2 className="slds-text-heading_small">
            26 potential duplicate leads were found.{' '}
            <a href="#" onClick={e => e.preventDefault()}>
              Select Leads to Merge
            </a>
          </h2>
        </div>
      </Toast>
    ));
  it('renders an error toast', () =>
    matchesMarkup(
      <Toast type="error">
        <div className="slds-notify__content">
          <h2 className="slds-text-heading_small">
            26 potential duplicate leads were found.{' '}
            <a href="#" onClick={e => e.preventDefault()}>
              Select Leads to Merge
            </a>
          </h2>
        </div>
      </Toast>
    ));
  it('renders a warning toast', () =>
    matchesMarkup(
      <Toast type="warning">
        <div className="slds-notify__content">
          <h2 className="slds-text-heading_small">
            26 potential duplicate leads were found.{' '}
            <a href="#" onClick={e => e.preventDefault()}>
              Select Leads to Merge
            </a>
          </h2>
        </div>
      </Toast>
    ));
  it('renders a success toast', () =>
    matchesMarkup(
      <Toast type="success">
        <div className="slds-notify__content">
          <h2 className="slds-text-heading_small">
            26 potential duplicate leads were found.{' '}
            <a href="#" onClick={e => e.preventDefault()}>
              Select Leads to Merge
            </a>
          </h2>
        </div>
      </Toast>
    ));
  it('renders a relative toast', () =>
    matchesMarkup(
      <Toast type="success" containerClassName="slds-is-relative">
        <UtilityIcon
          containerClassName="slds-m-right_small slds-no-flex slds-align-top"
          className="slds-icon_small"
          assistiveText={false}
          symbol="success"
        />
        <div className="slds-notify__content">
          <h2 className="slds-text-heading_small ">
            Account{' '}
            <a href="#" onClick={e => e.preventDefault()}>
              ACME - 100
            </a>{' '}
            widgets was created.
          </h2>
        </div>
      </Toast>
    ));
  it('renders a toast with an icon', () =>
    matchesMarkup(
      <Toast type="success">
        <UtilityIcon
          containerClassName="slds-m-right_small slds-no-flex slds-align-top"
          className="slds-icon_small"
          assistiveText={false}
          symbol="success"
        />
        <div className="slds-notify__content">
          <h2 className="slds-text-heading_small ">
            Account{' '}
            <a href="#" onClick={e => e.preventDefault()}>
              ACME - 100
            </a>{' '}
            widgets was created.
          </h2>
        </div>
      </Toast>
    ));
});
