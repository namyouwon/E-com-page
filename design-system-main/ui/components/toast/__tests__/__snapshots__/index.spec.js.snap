// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast renders a default toast for info 1`] = `
<div
  className="slds-notify_container"
>
  <div
    className="slds-notify slds-notify_toast slds-theme_info"
    role="status"
  >
    <span
      className="slds-assistive-text"
    >
      info
    </span>
    <div
      className="slds-notify__content"
    >
      <h2
        className="slds-text-heading_small"
      >
        26 potential duplicate leads were found.
         
        <a
          href="#"
          onClick={[Function]}
        >
          Select Leads to Merge
        </a>
      </h2>
    </div>
    <div
      className="slds-notify__close"
    >
      <button
        className="slds-button slds-button_icon slds-button_icon-inverse"
        onClick={[Function]}
        title="Close"
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon slds-button__icon_large"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Close
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Toast renders a relative toast 1`] = `
<div
  className="slds-notify_container slds-is-relative"
>
  <div
    className="slds-notify slds-notify_toast slds-theme_success"
    role="status"
  >
    <span
      className="slds-assistive-text"
    >
      success
    </span>
    <span
      className="slds-icon_container slds-icon-utility-success slds-m-right_small slds-no-flex slds-align-top"
      title="Description of icon when needed"
    >
      <svg
        aria-hidden={true}
        className="slds-icon slds-icon_small"
      >
        <use
          xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#success"
        />
      </svg>
    </span>
    <div
      className="slds-notify__content"
    >
      <h2
        className="slds-text-heading_small "
      >
        Account
         
        <a
          href="#"
          onClick={[Function]}
        >
          ACME - 100
        </a>
         
        widgets was created.
      </h2>
    </div>
    <div
      className="slds-notify__close"
    >
      <button
        className="slds-button slds-button_icon slds-button_icon-inverse"
        onClick={[Function]}
        title="Close"
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon slds-button__icon_large"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Close
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Toast renders a success toast 1`] = `
<div
  className="slds-notify_container"
>
  <div
    className="slds-notify slds-notify_toast slds-theme_success"
    role="status"
  >
    <span
      className="slds-assistive-text"
    >
      success
    </span>
    <div
      className="slds-notify__content"
    >
      <h2
        className="slds-text-heading_small"
      >
        26 potential duplicate leads were found.
         
        <a
          href="#"
          onClick={[Function]}
        >
          Select Leads to Merge
        </a>
      </h2>
    </div>
    <div
      className="slds-notify__close"
    >
      <button
        className="slds-button slds-button_icon slds-button_icon-inverse"
        onClick={[Function]}
        title="Close"
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon slds-button__icon_large"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Close
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Toast renders a toast with an icon 1`] = `
<div
  className="slds-notify_container"
>
  <div
    className="slds-notify slds-notify_toast slds-theme_success"
    role="status"
  >
    <span
      className="slds-assistive-text"
    >
      success
    </span>
    <span
      className="slds-icon_container slds-icon-utility-success slds-m-right_small slds-no-flex slds-align-top"
      title="Description of icon when needed"
    >
      <svg
        aria-hidden={true}
        className="slds-icon slds-icon_small"
      >
        <use
          xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#success"
        />
      </svg>
    </span>
    <div
      className="slds-notify__content"
    >
      <h2
        className="slds-text-heading_small "
      >
        Account
         
        <a
          href="#"
          onClick={[Function]}
        >
          ACME - 100
        </a>
         
        widgets was created.
      </h2>
    </div>
    <div
      className="slds-notify__close"
    >
      <button
        className="slds-button slds-button_icon slds-button_icon-inverse"
        onClick={[Function]}
        title="Close"
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon slds-button__icon_large"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Close
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Toast renders a warning toast 1`] = `
<div
  className="slds-notify_container"
>
  <div
    className="slds-notify slds-notify_toast slds-theme_warning"
    role="status"
  >
    <span
      className="slds-assistive-text"
    >
      warning
    </span>
    <div
      className="slds-notify__content"
    >
      <h2
        className="slds-text-heading_small"
      >
        26 potential duplicate leads were found.
         
        <a
          href="#"
          onClick={[Function]}
        >
          Select Leads to Merge
        </a>
      </h2>
    </div>
    <div
      className="slds-notify__close"
    >
      <button
        className="slds-button slds-button_icon slds-button_icon-inverse"
        onClick={[Function]}
        title="Close"
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon slds-button__icon_large"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Close
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Toast renders an error toast 1`] = `
<div
  className="slds-notify_container"
>
  <div
    className="slds-notify slds-notify_toast slds-theme_error"
    role="status"
  >
    <span
      className="slds-assistive-text"
    >
      error
    </span>
    <div
      className="slds-notify__content"
    >
      <h2
        className="slds-text-heading_small"
      >
        26 potential duplicate leads were found.
         
        <a
          href="#"
          onClick={[Function]}
        >
          Select Leads to Merge
        </a>
      </h2>
    </div>
    <div
      className="slds-notify__close"
    >
      <button
        className="slds-button slds-button_icon slds-button_icon-inverse"
        onClick={[Function]}
        title="Close"
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon slds-button__icon_large"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Close
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Toast renders an info toast 1`] = `
<div
  className="slds-notify_container"
>
  <div
    className="slds-notify slds-notify_toast slds-theme_info"
    role="status"
  >
    <span
      className="slds-assistive-text"
    >
      info
    </span>
    <div
      className="slds-notify__content"
    >
      <h2
        className="slds-text-heading_small"
      >
        26 potential duplicate leads were found.
         
        <a
          href="#"
          onClick={[Function]}
        >
          Select Leads to Merge
        </a>
      </h2>
    </div>
    <div
      className="slds-notify__close"
    >
      <button
        className="slds-button slds-button_icon slds-button_icon-inverse"
        onClick={[Function]}
        title="Close"
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon slds-button__icon_large"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Close
        </span>
      </button>
    </div>
  </div>
</div>
`;
